hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1209,11 +1212,97 @@ public void getMetricCall(TracingContext tracingContext) throws IOException {
             requestHeaders);
     try {
       op.execute(tracingContext);
+      sendReadMetrics(tracingContext);
     } finally {
       this.isSendMetricCall = false;
     }
   }
 
+  public void updateReadMetrics(String inputStreamId, int bufferLength, int requestedLength, long contentLength, long nextReadPos, boolean firstRead) {
+    if (abfsConfiguration.isReadCallsMetricEnabled()) {
+      readMetricData.append(currentTimeMillis()).append(COMMA)
+          .append(inputStreamId).append(COMMA)
+          .append(bufferLength).append(COMMA)
+          .append(requestedLength).append(COMMA)
+          .append(contentLength).append(COMMA)
+          .append(nextReadPos).append(COMMA)
+          .append(firstRead).append("\n");
+    }
+  }
+
+  public void sendReadMetrics(TracingContext tracingContext) throws IOException {
+    if (!abfsConfiguration.isReadCallsMetricEnabled() || readMetricData.length() == 0) {
```

Comment:
Since readMetricData is init with some string, length should not be 0. i think we can remove it.